### PR TITLE
components: Add Checkbox & Toggle

### PR DESCRIPTION
**Pull request by dimitrisnl**
Change-type: minor
Signed-off-by: Dimitrios Lytras <dimitrios@balena.io>

---
##### Contributor checklist

- [ ] I have rebuilt the README with `npm run build:docs`
- [ ] I have regenerated screenshots for any affected components with `npm run generate-screenshots`
- [ ] I have regenerated jest snapshots for any affected components with `npm run jest -- -u`

##### Reviewer Guidelines
- When submitting a review, please pick:
  - '*Approve*' if this change would be acceptable in the codebase (even if there are minor or cosmetic tweaks that could be improved).
  - '*Request Changes*' if this change would not be acceptable in our codebase (e.g. bugs, changes that will make development harder in future, security/performance issues, etc).
  - '*Comment*' if you don't feel you have enough information to decide either way (e.g. if you have major questions, or you don't understand the context of the change sufficiently to fully review yourself, but want to make a comment)
---

### CODE REVIEW

**Comment by dimitrisnl** [comment]
@balena-ci 	retest

**Comment by sradevski** [comment]
@dimitrisnl can you modify the styling to match the new hover styleguides? Check the PR I opened for the radio button

**Comment by dimitrisnl** [comment]
Thanks @sradevski 	I have updated the hover styles

File: README.md [hunk 1]
```diff
@@ -219,6 +220,24 @@ Section containing content and actions on the same topic.
 | `minHeight`  | `string` | `auto` | - | CSS minHeight property |
 | `children`  | `JSX.Element` | - | - | Any content that is internally wrapped in a Box |
 
+### Checkbox
+
+
+[View story source](https://github.com/balena-io-modules/rendition/blob/master/src/stories/Input.js)
+
+#### Props
```
Comment (sradevski) [inline README.md]: Maybe you can just link to the grommet documentation for anything that is not rendition-specific. Like this we avoid unnecessary duplication.

File: src/theme.ts [hunk 2]
```diff
@@ -252,4 +252,36 @@ export default {
...
+		},
+		hover: {
+			border: {
+				color: {
+					dark: colors.quartenary.dark,
+					light: colors.quartenary.dark,
+				},
+			},
+		},
+		toggle: {
```
Comment (sradevski) [inline src/theme.ts]: I am not a huge fan of the slow color animation on hover, if anything at least it can be made faster IMO. I am fine as it is though, just an opinion.

File: __tests__/components/__snapshots__/RadioButtonGroup.spec.js.snap [hunk 3]
```diff
@@ -143,7 +143,7 @@ exports[`RadioButtonGroup renders correctly 1`] = `
 
 .c4:hover input:not([disabled]) + div,
 .c4:hover input:not([disabled]) + span {
-  border-color: #527699;
+  border-color: #D1D8F2;
```
Comment (sradevski) [inline __tests__/components/__snapshots__/RadioButtonGroup.spec.js.snap]: Why did the radio get affected?

Comment (dimitrisnl) [reply __tests__/components/__snapshots__/RadioButtonGroup.spec.js.snap]: That's very strange indeed. taking a look

File: src/components/Terminal/XTermDefaultStyle.ts [hunk 4]
```diff
@@ -117,12 +117,4 @@ export default `
...
 	overflow: hidden;
 }
-
-.xterm-dim {
-	opacity: 0.5;
-}
-
-.xterm-underline {
-	text-decoration: underline;
-}
```
Comment (dimitrisnl) [inline src/components/Terminal/XTermDefaultStyle.ts]: @thgreasi 	Why are these changed? I didn't remove anything.

Comment (sradevski) [reply src/components/Terminal/XTermDefaultStyle.ts]: @dimitrisnl try pulling the latest master, remove node_modules, and see if you get the same result. We removed the package-lock.json, so that might be the culprit.

File: src/components/Checkbox.tsx [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+		font-family: ${props.theme.font};
+	}
+	& label input[checked] + div {
+		border-color: ${props.theme.colors.primary.main};
+		background: ${props.theme.colors.primary.main};
+		> svg {
+			stroke: #fff;
+		}
+	}
+	& label[disabled] {
```
Comment (sradevski) [inline src/components/Checkbox.tsx]: If I remember correctly, this can be set through the theme (not sure if it applies to the checkbox).

Comment (dimitrisnl) [reply src/components/Checkbox.tsx]: Unfortunately not 

- https://github.com/grommet/grommet/blob/master/src/js/components/CheckBox/StyledCheckBox.js#L13
- https://github.com/grommet/grommet/blob/master/src/js/components/CheckBox/StyledCheckBox.js#L13

Althought I don't see anything in the radio button types either.

Comment (sradevski) [reply src/components/Checkbox.tsx]: There is opacity in the `global` style, doesn't it apply to the checkbox? Ah ok I see I also used the same CSS, maybe it doesn't apply to these, otherwise there is a prop in `global.control.disabled` in the theme. Sorry for the noise.

File: src/components/Terminal/XTermDefaultStyle.ts (hunk 4 above)
Comment (dimitrisnl) [reply src/components/Terminal/XTermDefaultStyle.ts]: No change - is it safe to include? @sradevski
